OptimizelySDK/Optimizely.cs

```diff
@@ -429,10 +429,16 @@ public virtual bool IsFeatureEnabled(string featureKey, string userId, UserAttri
 
             if (decision != null)
             {
-                var experiment = Config.GetExperimentFromId(decision.ExperimentId);
+                // If experiment does not found in either experiment or rollout id map, return default value.
+                var experiment = Config.GetExperimentByIdFromAllExperiments(decision.ExperimentId);
+                if (!Validator.IsValidInstance(experiment))
```

Comment:
I personally don't think that we need a check here. 
getVariationForFeature will return a valid Decision object(with valid experiment and variation IDs) and you will have to write separate unit tests to cover this in code coverage. 

Just a suggestion: Rather than writing a separate method you can just check if the key is null. instanceValid somehow tells that experiment belongs to the Experiment class. 

The method should or a block of logic within a method should approach a single pattern. Either all conditions should have a single return point at the end. Or all of the conditions should return and there shouldn't be no else blocks. The logical code starting from:  if (decision != null) have both patterns. 

Another issue with returning from here is that you don't log anything in case you have invalid experiment/variation.